components/tasks/httpTask.brs

```diff
@@ -14,15 +14,15 @@ function handleCalls() as Object
                 response = fire(msg.getData())
             end if
         else if msgType = "roUrlEvent"
-                if msg.getInt() = 1
-                    if msg.getResponseCode() > 0
-                        response = handleResponse(msg.getString(),msg,true)
-                    else
-                        response = handleResponse(msg.getFailureReason(),msg,false)
-                    end if
+            if msg.getInt() = 1
+                if msg.getResponseCode() > 0
+                    response = handleResponse(msg.getString(), msg, true)
+                else
+                    response = handleResponse(msg.getFailureReason(), msg, false)
                 end if
-                m.top.response = response
-                return response
+            end if
+            m.top.response = response
+            return response
```

Comment:
Does this mean that we can handle only a single request? If yes then `m.urlRequest` AA is useless.

Reply:
I guess we'll save the request's result in the response's AA and after resolving it we'll clear it's index